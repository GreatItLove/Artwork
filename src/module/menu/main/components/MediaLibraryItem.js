import React from 'react';
import { ListItem } from 'material-ui/List';
import PropTypes from 'prop-types';

import { MEDIA_LIBRARY } from '../../../../config/routesPatch';

const propTypes = {
  style: PropTypes.object.isRequired,
  styleChild: PropTypes.object.isRequired
};

/**
 * @function MediaLibraryItem
 * @param {Object} style
 * @param {Object} styleChild
 * @returns {XML}
 */
function MediaLibraryItem({ style, styleChild }) {
  return (
    <ListItem
      key={MEDIA_LIBRARY}
      primaryText="Media Library"
      innerDivStyle={style}
      leftIcon={<i className="material-icons">video_library</i>}
      primaryTogglesNestedList={true}
      nestedItems={[
        <ListItem key="FileUploader" innerDivStyle={styleChild} primaryText="File Uploader" />,
        <ListItem key="View" innerDivStyle={styleChild} primaryText="View" />
      ]}
    />
  );
}

MediaLibraryItem.propTypes = propTypes;

export default MediaLibraryItem;
