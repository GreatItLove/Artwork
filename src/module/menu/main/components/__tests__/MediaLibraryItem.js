import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import MediaLibraryItem from '../MediaLibraryItem';

describe('MediaLibraryItem', () => {
  it('Snapshot of MediaLibraryItem', () => {
    const wrapper = shallow(<MediaLibraryItem style={{}} styleChild={{}} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
