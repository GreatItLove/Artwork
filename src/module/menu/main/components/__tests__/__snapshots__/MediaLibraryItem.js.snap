// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaLibraryItem Snapshot of MediaLibraryItem 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement="span"
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  innerDivStyle={Object {}}
  insetChildren={false}
  leftIcon={
    <i
      className="material-icons"
    >
      video_library
    </i>
  }
  nestedItems={
    Array [
      <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        innerDivStyle={Object {}}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="File Uploader"
        primaryTogglesNestedList={false}
        secondaryTextLines={1}
    />,
      <ListItem
        autoGenerateNestedIndicator={true}
        containerElement="span"
        disableKeyboardFocus={false}
        disabled={false}
        initiallyOpen={false}
        innerDivStyle={Object {}}
        insetChildren={false}
        nestedItems={Array []}
        nestedLevel={0}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onNestedListToggle={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        primaryText="View"
        primaryTogglesNestedList={false}
        secondaryTextLines={1}
    />,
    ]
  }
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="Media Library"
  primaryTogglesNestedList={true}
  secondaryTextLines={1}
/>
`;
